syntax = "proto3";

import "google/protobuf/empty.proto";

message XFDOString {
    string string = 1;
}

message XFDOPlayerAndPolicyNum {
    int64 player = 1;
    int64 policy_num = 2;
}

message XFDOPlayer {
    int64 player = 1;
}

message XFDOPolicySpecJson {
    string policy_spec_json = 1;
}

message XFDOPolicySpecList {
    repeated XFDOPolicySpecJson policy_spec_list = 1;
}

message XFDONewBestResponseParams {
    XFDOPolicySpecList metanash_specs_for_players = 1;
    repeated XFDOPolicySpecList delegate_specs_for_players = 2;
    int64 policy_num = 3;
}

message XFDOPolicyMetadataRequest {
    int64 player = 1;
    int64 policy_num = 2;
    string metadata_json = 3;
}

message XFDOConfirmation {
    bool result = 1;
}

message XFDOMetadata {
    string json_metadata = 1;
}

service XFDOManager {
    rpc GetLogDir (google.protobuf.Empty) returns (XFDOString) {}
    rpc GetManagerMetaData (google.protobuf.Empty) returns (XFDOMetadata) {}
    rpc ClaimNewActivePolicyForPlayer (XFDOPlayer) returns (XFDONewBestResponseParams) {}
    rpc SubmitFinalBRPolicy (XFDOPolicyMetadataRequest) returns (XFDOConfirmation) {}
    rpc IsPolicyFixed (XFDOPlayerAndPolicyNum) returns (XFDOConfirmation) {}
}